### ISSUE TITLE
name other_player == "Error: No unit"

### ISSUE THREAD

**Issue by aeroson** [michail-nikolaev/task-force-arma-3-radio#1096]
Sometimes for some client computers, the following is true.

```
{ name _x == "Error: No unit" } count allPlayers > 0
```

This results in some clients hearing some players as if they had TFAR disabled.
For the affected clients, this happens:

```
name bugged_player == "Error: No unit"
isPlayer bugged_player == false  // for this reason isPlayer should not be used
bugged_player in allPlayers == true // this should be used everywhere instead of isPlayer
```

TFAR thinks bugged_player is not a player and thus not sends his position to plugin.
So i recommend we make new function TFAR_fnc_isPlayer which uses _x in allPlayers and use that everywhere instead of isPlayer.

related https://github.com/michail-nikolaev/task-force-arma-3-radio/issues/95 ?

**Comment by commy2** [comment]
`name` is a very broken command in Arma 3. It will print error messages if the target unit is dead. It will also fail to report a valid name and print `"Error: No unit"` if the unit is dead for a certain amount of time or _when executed in the same frame the unit was created_
This last one could be the cause of this problem

**Comment by aeroson** [comment]
In rare cases it seems that for some computers some players are bugged in this way:
`name bugged_player == "Error: No unit"`  regardless of position, distance, respawn, time.
This happens during entire game session.
It seems to appear after `bugged_player` respawns.
It seems `bugged_player` can temporary fix it (until he respawns) by rejoining the server.
`bugged_player` is always fresh, taken from `allPlayers`.

**Comment by commy2** [comment]
As I said, the `name` command cannot be used in the frame the unit is created. This includes respawning.
The best would be to completely avoid relying on `name` (or `vehicleVarName` for that matter) in scripting and only use object references.

**Comment by aeroson** [comment]
`name` command was used after the unit was created (unless one frame takes up several minutes, which is not the case since TFAR did change audio position of other players), the `name` command was used few times over several seconds, with same result, the issue here is not `name` command

The issue is that every player in `allPlayers` command should (in my opinion) return true if used with `isPlayer` command, which is not the case, some of the `allPlayers` are actually not players according to `isPlayer`, that is the problem iam talking about.

This does not happen just one single frame, it does hapen during entire 2 hour game.

**Comment by aeroson** [comment]
Better report: this happens on Cloudstalker's PC, his PC just thinks Dennes is not a player, yet he is in allPlayers. The array was executed (on Cloudstalker's PC) then stored into setVeriable and then shown on my PC.
```
_x	=>	B Alpha 2-3:5 (Denness [AM 2FL]) REMOTE  
_x in allPlayers	=>	true
_x in allUnits	=>	false
_x in playableUnits	=>	false
_x in (allMissionObjects "")	=>	true
typeof _x	=>	"B_Soldier_TL_F"
str _x	=>	"def1e040# 1686468: ia_soldier_01.p3d REMOTE"
name _x	=>	"Error: No unit"
group _x	=>	<NULL-group>
side _x	=>	WEST
uniform _x	=>	"AOR2_Camo"
weapons _x	=>	["rhs_weap_hk416d10_m320","rhs_weap_m72a7","Laserdesignator_03"]
alive _x	=>	true
isPlayer _x	=>	false
getPlayerUID _x	=>	""
backpack _x	=>	"B_Parachute"
vest _x	=>	"AOR2_Vest_1"
netId _x	=>	"32:76"
getPos _x	=>	[7903.53,13361.6,1.22747]
owner _x	=>	0
```
Notice the ```isPlayer _x``` and ```_x in allPlayer```, it makes no sense.

**Comment by dedmen** [comment]
_x in allPlayers can be a big performance problem with many players.

**Comment by dedmen** [comment]
Still experiencing this?

**Comment by aeroson** [comment]
unknown,
didn't create any ticket in Arma bugtracker,
on public servers on rare occasions I can still see "Error: No unit" for some players, so I guess it has not been fixed

**Comment by kripto202** [comment]
Recently been experiencing some people having error: no unit and they have to reload to have this fixed. There are others that doesn't see the issue and can talk to him normal. The ones that are affected, can't hear the person at all.

**Comment by aeroson** [comment]
The two players that had this happen between them were located almost on the opposites of earth.
I think this has to do something with how Arma 3 optimizes it's network usage, like if the ping is too high and if the player is out of some range, it sends reduced player information.

**Comment by commy2** [comment]
>I think this has something to do with how Arma 3 optimizes it's network usage, like if the ping is too high and if the players are too far from each other ingame, it sends reduced player information.

No.

**Comment by dedmen** [comment]
Fixed in Arma 2.02
